.travis.yml

```diff
@@ -31,7 +31,10 @@ jobs:
       script: sudo env "PATH=$PATH" python3 -m test --case timezone --build-pipeline samples/build-from-yum.json
     - name: firewall
       before_install: sudo apt-get install -y systemd-container yum tar
-      script: sudo env "PATH=$PATH" python3 -m test --case firewall --build-pipeline samples/build-from-yum.json  
+      script: sudo env "PATH=$PATH" python3 -m test --case firewall --build-pipeline samples/build-from-yum.json
     - name: locale
       before_install: sudo apt-get install -y systemd-container yum tar
-      script: sudo env "PATH=$PATH" python3 -m test --case locale --build-pipeline samples/build-from-yum.json  
+      script: sudo env "PATH=$PATH" python3 -m test --case locale --build-pipeline samples/build-from-yum.json
+    - name: stage-tests
+      before_install: sudo apt-get install -y systemd-container yum
+      script: sudo env "PATH=$PATH" "OSBUILD_TEST_BUILD_PIPELINE=samples/base-from-yum.json" python3 -m unittest test.test_stages
```

Comment:
Please use `samples/build-from-yum.json` instead. It's quicker. I've removed the other one, because it's confusing :)

Reply:
What do you mean by quicker? When I'm using `base-from-yum` the pipeline works like this:

1) build f27
2) build f30
3) build f30

On the contrary with `build-from-yum`:

1) build f27
2) build f30

I think the first pipeline should be used, it sounds more reasonable to build the final tree by the same Fedora version. Currently, this is solved by using build step (F30) in each of test pipelines AND using build-pipelined argument (F27). IMO, this is confusing (which build pipeline goes first?) and it means I need to put the buildstep in each test pipeline, which is cluttering them a lot.

What do you think?

Reply:
I meant it's quicker because it has one stage instead of two. 

The pipeline you're building *always* needs to include its build pipeline, because its hash must include all the tools that were used to assemble the final image. The `--build-pipeline` parameter is there to get a host system into something that we know can produce the build pipeline. It is always run first. The Sorry if this is confusing.

The first case is indeed the one that we should use (for the reason you mention). The external pipeline makes a f27, to make it possible to build the "real" pipeline (f30 built on f30) on Ubuntu.

Does that make sense?

Reply:
I thought that `--build-pipeline` gets also included into the tree id. But I was wrong, thanks for the explanation. :) Therefore I added the build stage to all the test pipelines. :)